smithy-model/src/main/java/software/amazon/smithy/model/Model.java

```diff
@@ -761,6 +762,26 @@ public <T extends Shape> Stream<T> shapes(Class<T> shapeType) {
         return toSet(shapeType).stream();
     }
 
+    /**
+     * Gets a stream of shapes that have mixins.
+     *
+     * @return stream of shapes
+     */
+    public Stream<Shape> shapesWithMixins() {
+        return shapeMap.values().stream()
+            .filter(shape -> !shape.isMemberShape())
+            .filter(shape -> !shape.getMixins().isEmpty());
+    }
+
+    /**
+     * Gets a stream of shapes defined as mixins.
+     *
+     * @return stream of shapes
+     */
+    public Stream<Shape> mixins() {
```

Comment:
I think a much better way for people to get this and any other set of shapes with a trait is: `model.getShapesWithTrait(MixinTrait.class)`. It's cached too. I'd rather we keep it to that single way to do it since it's already ingrained everywhere else.

Reply:
Resolved in commit: 40775d9